index.js

```diff
@@ -22,15 +23,20 @@ var toString = Function.prototype.toString;
 var functionNameMatch = /\s*function(?:\s|\s*\/\*[^(?:*\/)]+\*\/\s*)*([^\s\(\/]+)/;
 function getFuncName(aFunc) {
   var name = '';
-  if (typeof Function.prototype.name === 'undefined' && typeof aFunc.name === 'undefined') {
-    // Here we run a polyfill if Function does not support the `name` property and if aFunc.name is not defined
-    var match = toString.call(aFunc).match(functionNameMatch);
-    if (match) {
-      name = match[1];
+  var isFunction = aFunc instanceof Function;
+  if (isFunction) {
```

Comment:
Maybe we should early return for non-functions? Like `if (typeof aFunc != 'function') return null`. Otherwise nesting is rather deep.